File: samples/javascript/pubsub/subscribe.js
const WebSocket = require('ws');
const { WebPubSubServiceClient } = require('@azure/web-pubsub');

async function main() {
  const hub = "pubsub";
  let serviceClient = new WebPubSubServiceClient(process.env.WebPubSubConnectionString, hub);
  let token = await serviceClient.getClientAccessToken();
  let ws = new WebSocket(token.url);
  ws.on('open', () => console.log('connected'));
  ws.on('message', data => console.log('Message received: %s', data));
}

main();

File: samples/python/pubsub/subscribe.py
import asyncio
import sys
import threading
import time
import websockets
from azure.messaging.webpubsubservice import (
    WebPubSubServiceClient
)


async def connect(url):
    async with websockets.connect(url) as ws:
        print('connected')
        while True:
            print(await ws.recv())

if len(sys.argv) != 3:
    print('Usage: python subscribe.py <connection-string> <hub-name>')
    exit(1)

connection_string = sys.argv[1]
hub_name = sys.argv[2]

client = WebPubSubServiceClient.from_connection_string(connection_string)
token = client.get_client_access_token(hub_name)

try:
    asyncio.get_event_loop().run_until_complete(connect(token['url']))
except KeyboardInterrupt:
    pass
